pkg/cmd/experimental/syncgroups/cli/syncgroups.go

```diff
@@ -156,82 +157,81 @@ func NewCmdSyncGroups(name, fullName string, f *clientcmd.Factory, out io.Writer
 	return cmd
 }
 
-type SyncBuilder interface {
-	GetGroupLister() (interfaces.LDAPGroupLister, error)
-	GetGroupNameMapper() (interfaces.LDAPGroupNameMapper, error)
-	GetUserNameMapper() (interfaces.LDAPUserNameMapper, error)
-	GetGroupMemberExtractor() (interfaces.LDAPMemberExtractor, error)
-}
-
 func (o *SyncGroupsOptions) Complete(typeArg, whitelistFile, blacklistFile, configFile string, args []string, f *clientcmd.Factory) error {
 	switch typeArg {
 	case string(GroupSyncSourceLDAP):
 		o.Source = GroupSyncSourceLDAP
 	case string(GroupSyncSourceOpenShift):
 		o.Source = GroupSyncSourceOpenShift
-
 	default:
 		return fmt.Errorf("unrecognized --type %q; allowed types %v", typeArg, strings.Join(AllowedSourceTypes, ","))
 	}
 
-	// if args are given, they are OpenShift Group names forming a whitelist
-	if len(args) > 0 {
-		o.Whitelist = append(o.Whitelist, args...)
+	var err error
+	o.Whitelist, err = buildNameList(o.Source, args, whitelistFile)
+	if err != nil {
+		return err
+	}
+	o.Blacklist, err = buildNameList(o.Source, []string{}, blacklistFile)
+	if err != nil {
+		return err
 	}
 
-	// unpack whitelist file from source
-	if len(whitelistFile) != 0 {
-		whitelistData, err := readLines(whitelistFile)
-		if err != nil {
-			return err
-		}
-		o.Whitelist = append(o.Whitelist, whitelistData...)
+	o.Config, err = decodeSyncConfigFromFile(configFile)
+	if err != nil {
+		return err
+	}
 
-		if o.Source == GroupSyncSourceOpenShift {
-			o.Whitelist, err = openshiftGroupNamesOnlyList(o.Whitelist)
-			if err != nil {
-				return err
-			}
-		}
+	osClient, _, err := f.Clients()
+	if err != nil {
+		return err
+	}
+	o.GroupInterface = osClient.Groups()
+
+	return nil
+}
+
+// buildNameLists builds a list from file and args
+func buildNameList(source GroupSyncSource, args []string, file string) ([]string, error) {
+	var list []string
+	if len(args) > 0 {
+		list = append(list, args...)
 	}
 
-	// unpack blacklist file from source
-	if len(blacklistFile) != 0 {
-		blacklistData, err := readLines(blacklistFile)
+	// unpack file from source
+	if len(file) != 0 {
+		listData, err := readLines(file)
 		if err != nil {
-			return err
+			return nil, err
 		}
-		o.Blacklist = append(o.Blacklist, blacklistData...)
+		list = append(list, listData...)
+	}
 
-		if o.Source == GroupSyncSourceOpenShift {
-			o.Blacklist, err = openshiftGroupNamesOnlyList(o.Blacklist)
-			if err != nil {
-				return err
-			}
+	if source == GroupSyncSourceOpenShift {
```

Comment:
refactor.  There should be two methods: `buildNameList` and `buildOpenShiftGroupNameList`.  The latter calls the former and runs the filter.